### PR TITLE
Feat: add harbor addon for image registry

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: experimental/addons/harbor/template.cue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+					clusters: parameter.clusters
+				}
+				if parameter.clusters == _|_ {
+					clusters: ["local"]
+				}
+			}
+		}]
+		workflow: steps: [{
+			type: "deploy"
+			name: "deploy-ck"
```
Comment (chivalryq) [inline experimental/addons/harbor/template.cue:28]: What is "deploy-ck"?

File: experimental/addons/harbor/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Harbor
+
+[Harbor](https://goharbor.io/) is an open source registry that secures artifacts with policies and role-based access control, ensures images are scanned and free from vulnerabilities, and signs images as trusted.
+
+This addon install harbor by helm chart. 
+
+:::caution
+This is just a experimental addon that it didn't configure persistent storage for harbor, please make sure you're just using it for demo.
+:::
```
Comment (chivalryq) [inline experimental/addons/harbor/README.md:9]: This syntax is not supported by vanilla markdown. Please just use ">"

Comment (wonderflow) [reply experimental/addons/harbor/README.md:9]: We should support it in velaux

File: experimental/addons/harbor/template.cue (hunk 1 above)
Comment (wonderflow) [reply experimental/addons/harbor/template.cue:28]: fixed

**Review by wangyikewxgm — APPROVED** [review]

File: experimental/addons/harbor/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+```
+kubectl get nodes -owide
+```
+
+The EXTERNAL-IP is what we need here as "<your-node-ip>".
+
+3. Install addon from experimental registry
+
+```
+vela addon enable experimental/harbor serviceType=nodePort externalURL=http://<your-node-ip>:30002
```
Comment (barnettZQG) [inline experimental/addons/harbor/README.md:30]: Is the NodePort number fixed?

Comment (wonderflow) [reply experimental/addons/harbor/README.md:30]: yes

**Pull request merged**